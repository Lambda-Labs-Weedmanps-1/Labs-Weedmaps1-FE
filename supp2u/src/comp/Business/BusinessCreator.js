import React, { useState } from 'react'
import BusinessAdd from './BusinessAdd'
import Axios from "axios";

import './business-creator.sass'

function BusinessCreator() {

    // The useState hook that will store the Business information
    const [businessInformation, setBusinessInformation] = useState([{
        "user_id": 0,
        "name": "",
        "website": "",
        "city": "",
        "state": "",
        "street": "",
        "zipcode": 0,
        "building_number": 420,
        "theme": "",
        "description": "",
        "hours": "",
        "recommended": null,
        "long": null,
        "lat": null}]);
    
        // hard coded user for test reasons
    let user = 1

    const changeHandler = event => {
        setBusinessInformation({ ...businessInformation, [event.target.name]: event.target.value });
      };

    //   submit form function
      const submit = e =>{
          e.preventDefault()
          console.log('new data', businessInformation)
                setTimeout(() => {
                Axios.post(`https://supp2udev.herokuapp.com/api/v1/users/${user}/businesses`)
            .then((res, req) => {
    
                res.data = businessInformation
                    
        
            }).catch(error =>{
            console.log('ERROR POST\n',error);
        }, 5000);
      })}

    
    //   JSX for BusinessCreator component
    return (
        <>
    <div className="form"> 
    <h3>Create your business</h3>
        <form onSubmit={submit}>

            <div className="input-box-type1">
                <label>Name of your business</label>
                <input
                    type="text"
                    name="name"
                    value={businessInformation.name}
                    placeholder="business name..." 
                    onChange={changeHandler}
                    />
            ....
            </div>
            <br/>
            {/* Here we bring in the address input, this is off in another component because of the extra frontend logic used to turn the address into a geocoordanate */}
            <BusinessAdd />
            <div className="input-box-type1">
                <label>Building number</label>
                <input
                    placeholder="Enter building number..."
                    type="integer"
                    name="building_number"
                    value={businessInformation.building_number}
                    onChange={changeHandler}
                     />
            </div>
            <div className="input-box-type1">
                <label>zipcode</label>
                <input
                    placeholder="zipcode..."
                    type="integer"
                    name="zipcode"
                    value={businessInformation.zipcode}
                    onChange={changeHandler} />
            </div>
            <div className="input-box-type1">
                <label>City</label>
                <input
                    placeholder="City name..." 
                    type="text"
                    name="city"
                    value={businessInformation.city}
                    onChange={changeHandler}/>
            </div>
            
            <div className="input-box-type1">
                <label>Description</label>
                <textarea
                    placeholder="description of resturant..."
                    type="text"
                    name="description"
                    value={businessInformation.description}
                    onChange={changeHandler} />
            </div>

            {/* maybe turn this into a drop down */}
            <div className="input-box-type1">
                <label>Theme</label>
                <input
                    placeholder="Enter resturant theme..."
                    type="text"
                    name="theme"
                    value={businessInformation.theme}
                    onChange={changeHandler} />
            </div>
            
            {/* maybe turn this into a form of its own */}
            <div className="input-box-type1">
                <label>Hours of Operations</label>
                <input
                    placeholder="Enter open and close hours..." 
                    type="text"
                    name="hours"
                    value={businessInformation.hours}
                    onChange={changeHandler}/>
            </div>
            <div className="input-box-type1">
                <label>Website</label>
                <input
                    placeholder="Enter website..."
                    type="text"
                    name="website"
                    value={businessInformation.website}
                    onChange={changeHandler} />
            </div>


            <button> ................... </button>

        </form>
    </div>
    </>
    )
}

export default BusinessCreator
