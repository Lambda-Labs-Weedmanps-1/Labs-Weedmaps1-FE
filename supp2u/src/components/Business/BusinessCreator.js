import React, { useState } from 'react'
import Axios from "axios";

import './business-creator.sass'

function BusinessCreator() {

    // The useState hook that will store the Business information
    const [businessInformation, setBusinessInformation] = useState([{
        "user_id": 0,
        "name": "",
        "website": "",
        "city": "",
        "state": "",
        "street": " ",
        "zipcode": 0,
        "building_number": 420,
        "theme": "",
        "description": "",
        "hours": "",
        "recommended": null,
        "long": "",
        "lat": ""}]);
    
        // hard coded user for test reasons
    let user = 1
    const changeHandler = event => {
        setBusinessInformation({ ...businessInformation, [event.target.name]: event.target.value });
    };
    
    //   submit form function
    const submit = e =>{
        e.preventDefault()
        // Transmutes the address into a useable array
        if(businessInformation.street === undefined){console.log('Address in Required')}else{
        const newAddress = businessInformation.street
        const newAddressArray = newAddress.split(' ')
        //   address look up function
        // axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&key=XXXXXXXXXXXXXXXXXXX`)
        Axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=${newAddressArray[0]}+${newAddressArray[1]}+${newAddressArray[2]},+${businessInformation.city},+${businessInformation.state}&key=${process.env.REACT_APP_GCOORDINATES}`)
        .then (res => {
        //     // sends location to businessInformation
           businessInformation.lat = res.data.results[0].geometry.location.lat.toString()
           businessInformation.long = res.data.results[0].geometry.location.lng.toString() 
        })
        console.log('data to be sent to backend', businessInformation)
        
            Axios.post(`http://localhost:3000/api/v1/users/1/businesses`, businessInformation)
            .then((res, req) => { console.log('sent') })
            .catch(error =>{console.log('ERROR POST\n',error);
        });
    }}

    //   JSX for BusinessCreator component
    return (
        <>
    <h3>Create your business</h3>
    <div className="form"> 
        <form onSubmit={submit}>

            <div className="input-box-type1">
                <label>Name of business <span className="required-span">*</span></label>
                <input
                    type="text"
                    name="name"
                    value={businessInformation.name}
                    placeholder="business name..." 
                    onChange={changeHandler}
                    />
            </div>
            <br/>
            
            <div className="input-box-type1">
                <label>Address <span className="required-span">*</span></label>
                <input
                        type="text"
                        name="street"
                        placeholder="address"
                        onChange={changeHandler}
                        value={businessInformation.street}
                    />
            </div>
            
            <div className="input-box-type1">
                <label>Building number </label>
                <input
                    placeholder="Enter building number..."
                    type="integer"
                    name="building_number"
                    value={businessInformation.building_number}
                    onChange={changeHandler}
                     />
            </div>
            <div className="input-box-type1">
                <label>City <span className="required-span">*</span></label>
                <input
                    placeholder="City name..." 
                    type="text"
                    name="city"
                    value={businessInformation.city}
                    onChange={changeHandler}/>
            </div>
            <div className="input-box-type1">
                <label>State <span className="required-span">*</span></label>
                <input
                        type="text"
                        name="state"
                        placeholder="Enter state"
                        onChange={changeHandler}
                        value={businessInformation.state}
                    />
            </div>
            <div className="input-box-type1">
                <label>zipcode</label>
                <input
                    placeholder="zipcode..."
                    type="integer"
                    name="zipcode"
                    value={businessInformation.zipcode}
                    onChange={changeHandler} />
            </div>
            
            <div className="input-box-type1">
                <label>Description</label>
                <textarea
                    placeholder="description of resturant..."
                    type="text"
                    name="description"
                    value={businessInformation.description}
                    onChange={changeHandler} />
            </div>

            {/* maybe turn this into a drop down */}
            <div className="input-box-type1">
                <label>Theme</label>
                <input
                    placeholder="Enter resturant theme..."
                    type="text"
                    name="theme"
                    value={businessInformation.theme}
                    onChange={changeHandler} />
            </div>
            
            {/* maybe turn this into a form of its own */}
            <div className="input-box-type1">
                <label>Hours of Operations</label>
                <input
                    placeholder="Enter open and close hours..." 
                    type="text"
                    name="hours"
                    value={businessInformation.hours}
                    onChange={changeHandler}/>
            </div>
            <div className="input-box-type1">
                <label>Website</label>
                <input
                    placeholder="Enter website..."
                    type="text"
                    name="website"
                    value={businessInformation.website}
                    onChange={changeHandler} />
            </div>
            <span className="required-span">* required</span>


            <button className="create-business-button"> Create Business </button>

        </form>
    </div>
    </>
    )
}

export default BusinessCreator
