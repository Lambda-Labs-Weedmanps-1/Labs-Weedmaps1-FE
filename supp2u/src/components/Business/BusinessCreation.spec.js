import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import Enzyme from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';

import BusinessCreator from './BusinessCreator'

configure({ adapter: new Adapter() });

describe('Business Creator Renders', () => {
    // If Map component render is ever non functional, this should fail
    it('should render correctly when component called', () => {
      const Business = shallow(<BusinessCreator />);

      expect(Business).toMatchSnapshot();
    });
});

describe('input values for business creation should be present', () => {
    const container = shallow(<BusinessCreator />)

    // it('onchange should work', () => {
    //     let container = shallow(<BusinessCreator />)
    //     container.find('input[id="test1"]').simulate('change', {
    //         target: {
    //             value:'testpassword'
    //         }
    //     });
    //     expect(container.find('input[name="name"]').prop('value')).toBe(
    //         'testpassword'
    //     );
    // });

    it('should have a input field', () => {
        expect(container.find('input[type="text"]').length).toEqual(6)
    })
});

