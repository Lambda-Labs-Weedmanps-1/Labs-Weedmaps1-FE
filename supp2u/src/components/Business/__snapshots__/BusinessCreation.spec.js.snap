// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business Creator Display should render correctly when component called 1`] = `
<Fragment>
  <h3>
    Create your business
  </h3>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-box-type1"
      >
        <label>
          Name of business 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          id="test1"
          name="name"
          onChange={[Function]}
          placeholder="business name..."
          type="text"
        />
      </div>
      <br />
      <div
        className="input-box-type1"
      >
        <label>
          Address 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="street"
          onChange={[Function]}
          placeholder="address"
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Building number 
        </label>
        <input
          name="building_number"
          onChange={[Function]}
          placeholder="Enter building number..."
          type="integer"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          City 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="city"
          onChange={[Function]}
          placeholder="City name..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          State 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="state"
          onChange={[Function]}
          placeholder="Enter state"
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          zipcode
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="zipcode"
          onChange={[Function]}
          placeholder="zipcode..."
          type="integer"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Description
        </label>
        <textarea
          name="description"
          onChange={[Function]}
          placeholder="description of resturant..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Theme
        </label>
        <input
          name="theme"
          onChange={[Function]}
          placeholder="Enter restaurant theme..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Website
        </label>
        <input
          name="website"
          onChange={[Function]}
          placeholder="Enter website..."
          type="text"
        />
      </div>
      <span
        className="required-span"
      >
        * required
      </span>
      <ImageUploader
        selectImage={[Function]}
        unselectImage={[Function]}
      />
      <button
        className="create-business-button"
      >
         Create Business 
      </button>
    </form>
  </div>
</Fragment>
`;

exports[`Business Creator Renders should render correctly when component called 1`] = `
<Fragment>
  <h3>
    Create your business
  </h3>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-box-type1"
      >
        <label>
          Name of business 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          id="test1"
          name="name"
          onChange={[Function]}
          placeholder="business name..."
          type="text"
        />
      </div>
      <br />
      <div
        className="input-box-type1"
      >
        <label>
          Address 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="street"
          onChange={[Function]}
          placeholder="address"
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Building number 
        </label>
        <input
          name="building_number"
          onChange={[Function]}
          placeholder="Enter building number..."
          type="integer"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          City 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="city"
          onChange={[Function]}
          placeholder="City name..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          State 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="state"
          onChange={[Function]}
          placeholder="Enter state"
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          zipcode
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="zipcode"
          onChange={[Function]}
          placeholder="zipcode..."
          type="integer"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Description
        </label>
        <textarea
          name="description"
          onChange={[Function]}
          placeholder="description of resturant..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Theme
        </label>
        <input
          name="theme"
          onChange={[Function]}
          placeholder="Enter restaurant theme..."
          type="text"
        />
      </div>
      <div
        className="input-box-type1"
      >
        <label>
          Website
        </label>
        <input
          name="website"
          onChange={[Function]}
          placeholder="Enter website..."
          type="text"
        />
      </div>
      <span
        className="required-span"
      >
        * required
      </span>
      <ImageUploader
        selectImage={[Function]}
        unselectImage={[Function]}
      />
      <button
        className="create-business-button"
      >
         Create Business 
      </button>
    </form>
  </div>
</Fragment>
`;
